ElfUtils/ElfFile.cpp

```diff
@@ -27,8 +27,8 @@ class ElfFileImpl : public ElfFile {
       std::string_view file_path,
       llvm::object::OwningBinary<llvm::object::ObjectFile>&& owning_binary);
 
-  outcome::result<ModuleSymbols, std::string> LoadSymbols() const override;
-  outcome::result<uint64_t, std::string> GetLoadBias() const override;
+  Result<ModuleSymbols, ErrorMessage> LoadSymbols() const override;
```

Comment:
Should we have a type like 

ErrorOr<ModuleSymbols>

? 

Adding ErrorMessage to every result that can have an error seems more clutter than necessary.

Reply:
It might get confusing for the following case:

ErrorOr<void> result = DoSomething();

if (!result) {
  // now in Result<T,E> this is false if there was an error, should it be also false for ErrorOr?
}

Reply:
Having said that this can be resolved by explicitly calling has_failure() instead of implicitly converting to boolean.

Reply:
I would be in favor of that. Unless there is a good reason why not

Reply:
If we want to do this, in my opinion it should be `ErrorMessageOr`. Yes, it's longer and then maybe not that different from `Result<..., ErrorMessage>`, but having `ErrorMessage` as the error type is yes the most common case, but it's not the only case (as we said we could have enums for "recoverable" errors), and `ErrorOr` hides that.

Reply:
Yes, agree, it should be ErrorMessageOr if we do this.

Reply:
This said, in general I find the inversion with `Error` in front weird: it's like an error is the "main" return type.

Reply:
@dimitry- I don't understand your case. Why can/should ErrorMessageOr<T> not behave exactly like Result<T, ErrorMessage>?

Reply:
done.